auacm/app/modules/competition_manager/views.py

```diff
@@ -31,6 +32,82 @@ def get_competitions():
     })
 
 
+@app.route('/api/competitions', methods=['POST'])
+@login_required
+def create_competition():
+    data = request.form
+    if current_user.admin == 0:
+        return serve_error('Only admins can create competitions', 401)
+    if not data['name'] or not data['start_time'] or not data['length'] or\
+            not data['problems']:
+        return serve_error('You must specify name, startTime,' +
+                ' and length attributes')
+    competition = Competition(
+        name=data['name'],
+        start=int(data['start_time']),
+        stop=(int(data['start_time']) + int(data['length']))
+    )
+    competition.commit_to_session()
+
+    comp_problems = loads(data['problems'])
+    for problem in comp_problems:
+        session.add(CompProblem(
+            label=problem['label'],
```

Comment:
I guess I should have asked this last night, but this is set up so that the client picks the labels for the problems?